### PR TITLE
build(azure-local-service): esm & trim

### PR DESCRIPTION
**Pull request by DLehenbauer**
This is a leaf package that re-exposes Tinylicious as the "Azure Local Service" for development and testing.

Converted to ESM because it was easy.

Removed all exports since it's a service.

### CODE REVIEW

File: azure/packages/azure-local-service/package.json [hunk 1]
```diff
@@ -10,38 +10,13 @@
...
-	"main": "dist/index.js",
-	"types": "dist/index.d.ts",
-	"bin": "dist/index.js",
+	"type": "module",
+	"bin": "lib/index.js",
 	"scripts": {
 		"build": "fluid-build . --task build",
 		"build:compile": "fluid-build . --task compile",
-		"build:docs": "api-extractor run --local",
+		"build:docs": "fluid-build . --task api",
```
Comment (jason-ha) [inline azure/packages/azure-local-service/package.json:18]: There is no api task. Also don't need api-extractor files.

File: azure/packages/azure-local-service/tsconfig.cjs.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jason-ha) [inline azure/packages/azure-local-service/tsconfig.cjs.json:1]: Unused file

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: azure/packages/azure-local-service/package.json (hunk 1 above)
Comment (DLehenbauer) [reply azure/packages/azure-local-service/package.json:20]: @jason-ha, @tylerbutler - policy-check won't let me remove build:docs...  so add an api task?

Comment (jason-ha) [reply azure/packages/azure-local-service/package.json:20]: I think others have that - or you can add an exception.
api-extractor:commonjs can't remain.

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: azure/packages/azure-local-service/package.json (hunk 1 above)
Comment (jason-ha) [reply azure/packages/azure-local-service/package.json]: In folder api-extractor files are still sitting around.

**Review by jason-ha — CHANGES_REQUESTED** [review]
unused api-extractor files should be removed

**Review by jason-ha — APPROVED** [review]

**Pull request merged**